crates/forge-runner/src/gas/report.rs

```diff
@@ -0,0 +1,119 @@
+use crate::gas::stats::GasStats;
+use cheatnet::trace_data::{CallTrace, CallTraceNode};
+use debugging::ContractName as DebuggingContractName;
+use debugging::ContractsDataStore;
+use starknet_api::core::{ClassHash, EntryPointSelector};
+use starknet_api::execution_resources::GasVector;
+use std::collections::BTreeMap;
+
+type ContractName = String;
+type Selector = String;
+
+#[derive(Debug, Clone)]
+pub struct GasSingleTestInfo {
+    pub gas_used: GasVector,
+    pub report_data: ReportData,
+}
+
+#[derive(Debug, Clone)]
+pub struct ReportData(BTreeMap<ContractName, ContractInfo>);
+
+#[derive(Debug, Clone, Default)]
+pub struct ContractInfo {
+    pub gas_used: GasVector,
+    pub functions: BTreeMap<Selector, SelectorReportData>,
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct SelectorReportData {
+    pub gas_stats: GasStats,
+    pub n_calls: u64,
+    pub records: Vec<u64>,
+}
+
+impl GasSingleTestInfo {
+    #[must_use]
+    pub fn new(gas_used: GasVector) -> Self {
+        Self {
+            gas_used,
+            report_data: ReportData(BTreeMap::new()),
```

Comment:
nit: Seems more idiomatic (need to derive `Default` on `ReportData`):
```suggestion
            report_data: ReportData::default(),
```